packages/components/src/DataList/components/DataListHeaderTile/DataListHeaderTile.tsx

```diff
@@ -38,24 +67,41 @@ export function DataListHeaderTile<T extends DataListObject>({
     </Tag>
   );
 
-  function toggleSorting(sortingKey: string) {
+  function toggleSorting(sortingKey: string, order?: "asc" | "desc") {
     const isSameKey = sortingKey === sortingState?.key;
+    const isDescending = sortingState?.order === "desc";
 
-    if (isSameKey && sortingState?.order === "desc") {
+    if (isSameKey && isDescending && order !== "asc") {
+      if (order === "desc") return;
       sorting?.onSort(undefined);
 
       return;
     }
 
+    const sortingOrder = order || (isSameKey && !isDescending ? "desc" : "asc");
+
     sorting?.onSort({
-      order: isSameKey && sortingState?.order === "asc" ? "desc" : "asc",
       key: sortingKey,
+      order: sortingOrder,
     });
   }
 
   function handleOnClick() {
     if (!isSortable) return;
 
-    toggleSorting(headerKey);
+    if (sortableItem?.options) {
+      setIsDropDownOpen(!isDropDownOpen);
+    } else {
+      toggleSorting(headerKey);
+    }
+  }
+
+  function handleSelectChange(selectedOption: SortableOptions) {
+    if (sortableItem) {
+      toggleSorting(sortableItem.key, selectedOption.order);
```

Comment:
I think we should add the `sortableItem.label` to this call, too. For example, in the current DataList story, if you select "First name (A-Z)" or "Last name (A-Z)" the output of `sorting` will be the same: `{key: 'label', order: 'desc'}`.

Reply:
Also done here 8d28274470e494cf158f75a3183715965e3b1a01